src/objects.c

```diff
@@ -1526,12 +1526,15 @@ P11PROV_OBJ *p11prov_create_secret_key(P11PROV_CTX *provctx,
     CK_OBJECT_CLASS key_class = CKO_SECRET_KEY;
     CK_KEY_TYPE key_type = CKK_GENERIC_SECRET;
     CK_BBOOL val_true = CK_TRUE;
+    CK_BBOOL val_false = CK_FALSE;
     CK_BBOOL val_token = session_key ? CK_FALSE : CK_TRUE;
-    CK_ATTRIBUTE key_template[5] = {
+    CK_ATTRIBUTE key_template[7] = {
         { CKA_CLASS, &key_class, sizeof(key_class) },
         { CKA_KEY_TYPE, &key_type, sizeof(key_type) },
         { CKA_TOKEN, &val_token, sizeof(val_token) },
         { CKA_DERIVE, &val_true, sizeof(val_true) },
+        { CKA_SIGN, &val_true, sizeof(val_true) },
+        { CKA_PRIVATE, &val_false, sizeof(val_false) },
```

Comment:
Instead of adding these values to p11prov_create_secret_key() you should probably add a p11prov_create_mac_key function.

We do not want all of the secret keys to be enabled for signing unconditionally.
(and yes we should probably rename p11prov_create_secret_key -> p11prov_create_derive_key)